content/concepts/protocols.md

```diff
@@ -1,227 +1,167 @@
-
+This article is concerned with protocols that compose libp2p
+using core libp2p abstractions like the [transport](/concepts/transport), 
+[peer identity](/concepts/peers#peer-id/), [addressing](/concepts/addressing/) 
+abstractions to name a few. 
 
 ## What is a libp2p protocol?
 
-A libp2p protocol has these key features:
+A libp2p protocol is a network protocol that includes
+the key features mentioned below.
 
-#### Protocol Ids
+#### Protocol IDs
 
-libp2p protocols have unique string identifiers, which are used in the [protocol negotiation](#protocol-negotiation) process when connections are first opened.
+libp2p protocols have unique string identifiers, which are used in the 
+[protocol negotiation](#protocol-negotiation) process when connections are first opened.
 
-By convention, protocol ids have a path-like structure, with a version number as the final component:
+By convention, protocol IDs have a path-like structure, with a version number as the 
+final component:
 
 ```
 /my-app/amazing-protocol/1.0.1
 ```
 
-Breaking changes to your protocol's wire format or semantics should result in a new version
-number. See the [protocol negotiation section](#protocol-neotiation) for more information about
-how version selection works during the dialing and listening process.
+Breaking changes to a protocol's wire format or semantics should result in a new 
+version number. See the [protocol negotiation section](#protocol-neotiation) for more 
+information about how version selection works during the dialing and listening.
 
 {{% notice "info" %}}
 
-While libp2p will technically accept any string as a valid protocol id,
+While libp2p can accept any string as a valid protocol ID,
 using the recommended path structure with a version component is both
 developer-friendly and enables [easier matching by version](#match-using-semver).
 
 {{% /notice %}}
 
 #### Handler functions
 
-To accept connections, a libp2p application will register handler functions for protocols using their protocol id with the
-[switch][definition_switch] (aka "swarm"), or a higher level interface such as [go's Host interface](https://github.com/libp2p/go-libp2p-core/blob/master/host/host.go).
-
-The handler function will be invoked when an incoming stream is tagged with the registered protocol id.
-If you register your handler with a [match function](#using-a-match-function), you can choose whether
-to accept non-exact string matches for protocol ids, for example, to match on [semantic major versions](#match-using-semver).
+A libp2p application will register handler functions 
+for protocols using their protocol ID with the
+[switch][definition_switch] (aka "swarm"), or a higher level interface such as 
+[go's Host interface](https://github.com/libp2p/go-libp2p-core/blob/master/host/host.go) 
```

Comment:
That's very implementation-specific. Can we say that application will define a stream handler, that takes over the stream after protocol negotiation. Everything sent and received after this negotiation step is then defined by the application protocol.